libcache/cache.c

```diff
@@ -0,0 +1,391 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Cache library
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Malgorzata Wrobel
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "cache.h"
+
+#include <sys/list.h>
+// #include "list.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include <errno.h>
+#include <string.h>
+
+#define LIBCACHE_NUM_WAYS   4
+#define LIBCACHE_ADDR_WIDTH 64
+
+
+typedef struct cacheline_s cacheline_t;
+
+
+struct cacheline_s {
+	uint64_t tag;
+	uint32_t *data;
+	unsigned char isValid;
+	unsigned char isDirty;
```

Comment:
You can store both flags on one common unsigned char and use macro to access them
```suggestion
	unsigned char flags;
	
	(...)
	#define IS_VALID(f) ((f) & (1 << 0) != 0 ? 1 : 0)
	#define IS_DIRTY(f) ((f) & (1 << 1) != 0 ? 1 : 0)
	#define SET_VALID(f, state) do { (f) |= (1 << 0); } while (0) 
```
etc

Reply:
Sorry, I've meant
#define SET_VALID(f, state) do { (f) |= (1 << ((state) != 0 : 1 ; 0)); } while (0)

Reply:
Thank you, I've applied suggested changes